test/uk/gov/hmrc/channelpreferences/controllers/PreferenceControllerSpec.scala

```diff
@@ -60,8 +57,8 @@ class PreferenceControllerSpec extends PlaySpec with ScalaFutures with MockitoSu
             ec: ExecutionContext): Future[Either[Int, EmailVerification]] =
             Future.successful(Left(SERVICE_UNAVAILABLE))
         },
-        mockAuthConnector,
-        mockEntityResolverConnector,
+        mock[AuthConnector],
+        mock[EntityResolverConnector],
```

Comment:
Having taken a look at the code I would have prefered `TestSetup` to construct `PreferenceController` with a mock for `CdsPreference`, here: 
```
trait TestSetup {
    val controller = new PreferenceController(
      new CdsPreference {
        override def getPreference(c: Channel, enrolmentKey: String, taxIdName: String, taxIdValue: String)(
          implicit hc: HeaderCarrier,
          ec: ExecutionContext): Future[Either[Int, EmailVerification]] =
          Future.successful(Left(SERVICE_UNAVAILABLE))
      },
      mock[AuthConnector],
      mock[EntityResolverConnector],
      Helpers.stubControllerComponents()
    )
    ```

Reply:
I have pushed my thinking, wdyt?

Reply:
Ok,  @spifix-hmrc,

It seems to me that previous authors wanted to use stubs (instead of mocks) when injecting the `CdsPreference` collaborator into the _"thing under test"_ (see https://martinfowler.com/articles/mocksArentStubs.html for more details on this specific subject).

It also seems you're asking me to change unit tests that were written by others before me (and which do not strictly relate to the scope of the JIRA ticket assigned to me). 

That's ok, for now, of course! 

I just pushed another set of changes intended to retrofit the _"fresh-fixture"_ strategy and use mocks instead of stubs wherever possible ... but, please ... kindly speaking, be aware I am supposed to work on DC-3504 during this sprint. Thank you.

Reply:
I am asking you to improve the code quality rather than blindly follow what has gone before :)

Reply:
@spifix-hmrc better if we have a quick chat over the phone, please